### PR TITLE
feat: add application terminate

### PR DESCRIPTION
**Pull request by SonyPradana**

### CODE REVIEW

File: src/System/Integrate/Http/Karnel.php [hunk 1]
```diff
@@ -7,16 +7,17 @@
...
 use System\Http\Response;
+use System\Router\Router;
 
 class Karnel
 {
     /** @var Container */
     protected $app;
 
-    /** @var array<int, class-string> Global middleware */
+    /** @var array<int, class-string|callable> Global middleware */
```
Comment (anggerpradana) [inline src/System/Integrate/Http/Karnel.php:17]: we need anonymous class?

File: src/System/Integrate/Http/Karnel.php [hunk 2]
```diff
@@ -51,6 +52,22 @@ public function handle(Request $request)
         return $pipeline($request);
     }
 
+    /**
+     * Terminate Requesr and Response.
+     */
+    public function terminate(Request $request, Response $response): void
+    {
+        $middleware = $this->dispatcherMiddleware($request) ?? [];
```
Comment (anggerpradana) [inline src/System/Integrate/Http/Karnel.php:60]: why not directly call `Router::current()`

File: tests/Integrate/Http/KarnelTerminateTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            protected function dispatcherMiddleware(Request $request)
+            {
+                return [
+                    new class() {
+                        public function terminate(Request $request, Response $respone)
+                        {
+                            echo $request->getUrl();
+                            echo $respone->getContent();
+                        }
+                    },
```
Comment (anggerpradana) [inline tests/Integrate/Http/KarnelTerminateTest.php:40]: its work, but not track able if using anonymous class

**Review by anggerpradana — CHANGES_REQUESTED** [review]
we really need anonymous class?

File: src/System/Integrate/Http/Karnel.php (hunk 1 above)
Comment (SonyPradana) [reply src/System/Integrate/Http/Karnel.php:17]: you right, using anonymous will untracked middleware was run.

File: src/System/Integrate/Http/Karnel.php (hunk 2 above)
Comment (SonyPradana) [reply src/System/Integrate/Http/Karnel.php:60]: this helped to test terminate function intead mock a router dispatch. i think is well to use this function as long as default is route current. so we don't need overide latter.

**Review by anggerpradana — APPROVED** [review]

**Pull request merged**